### DISCUSSION TITLE
Memory Usage significant increase since 0.3.22

### DISCUSSION THREAD

**Discussion by TedLin1993** [open-webui/open-webui#5851] [General]
# Bug Report

## Description

**Bug Summary:**

When the version reaches or exceeds 0.3.22, memory usage will significantly increase from approximately 300MB to 600MB. This poses a substantial cost difference for serverless systems. I hope someone can identify the cause and resolve this issue.

## Reproduction Details

Environment Varaible:
- DATABASE_URL: postgresql://xxx 
- OPENAI_API_KEY: sk-xxx
- RAG_EMBEDDING_ENGINE: openai
- AUDIO_STT_ENGINE: openai 
- ENABLE_OLLAMA_API: false

**Steps to Reproduce:**

I deployed different versions of Open WebUI using Google Cloud Run and monitored the memory usage of each version. In version 0.3.21, the memory usage was around 300MB (with a set limit of 1GB)
![image](https://github.com/user-attachments/assets/5a795f4d-32de-445b-b8f7-bec9538e7ac9)

Whereas when the version was upgraded to 0.3.22, the memory usage surged to approximately 600MB:

![image](https://github.com/user-attachments/assets/30880d3e-4cd3-4a43-9f73-6384f001405c)

**Comment by justinh-rahb** [comment]
Try 0.3.30, that's the current latest.

**Comment by TedLin1993** [comment]
I have tried, the memory usage still significant high.

**Comment by TedLin1993** [comment]
I switch version from 0.3.22 to 0.3.30:

![image](https://github.com/user-attachments/assets/d19dcee2-fdd6-43aa-a676-965076faf796)

As you can see, the memory usage remains the same.

**Comment by wonrax** [comment]
try this, it helped me (note that you may need to change the settings in the admin panel because the environment variables does not have any effect after you already init the application)
https://docs.openwebui.com/tutorials/tips/reduce-ram-usage
